### ISSUE TITLE
Incomplete handling of VM status for un-registration

### ISSUE THREAD

**Issue by hferentschik** [minishift/minishift#318]
Currently the code for un-registrating the VM in _stop.go_ / _delete.go_ looks like this:

    if !drivers.MachineInState(host.Driver, state.Stopped)() {
        ...
    }

This means that un-registration will occur in all other states than stopped. However, there are other states in which un-regsitration won't work. We have the following states:

	"Running",
	"Paused",
	"Saved",
	"Stopped",
	"Stopping",
	"Starting",
	"Error",
	"Timeout",

Even more, the code for un-registration should not be in _stop.go_ / _delete.go_, but rather in _cluster.StopHost_ resp. _cluster.DeleteHost_, so that other callers of _StopHost_ can un-register as well.

**Comment by praveenkumar** [comment]
@hferentschik Agree on that behavior. In past we agreed on that piece of code should be independent of `cluster.go` file because that's something comes from minikube and it will be easily rebase if we need to get changes from minikube repo. If that is what we still follow then I think better we should handle all state in same code base instead making changes in `cluster.go`

**Comment by praveenkumar** [comment]
@hferentschik I think we only have to switch our condition to `if drivers.MachineInState(host.Driver, state.Running)()` because for any other state ssh command will not be able to execute on the VM side which handle unregister. wdyt?

**Comment by amitkrout** [comment]
Adding more execution result to it.
``
$ minishift.exe version
minishift version: 1.0.0-beta.2
Virtualbox : 5.1.10
Platform : Windows 10
```
Randomly i was executing sanity tests includes docker images pull and run within the box. Then i exit from the box and found the error while running minishift command includes ssh, console and stop

```
$ minishift.exe start --vm-driver=virtualbox --iso-url https://github.com/minishift/minishift-centos-iso/releases/download/v1.0.0-beta.1/minishift-centos.iso
Starting local OpenShift instance using 'virtualbox' hypervisor...
Downloading ISO
 316.00 MB / 316.00 MB  100.00% 0sssss
E0123 10:07:53.048934    3936 start.go:135] Error starting machine: Error creating. Error creating machine: Error checking the host: Error checking and/or regenerating the certs: There was an error validating certificates for host "192.168.99.100:2376": dial tcp 192.168.99.100:2376: i/o timeout
You can attempt to regenerate them using 'docker-machine regenerate-certs [name]'.
Be advised that this will trigger a Docker daemon restart which might stop running containers.
. Retrying.
 18.42 MB / 18.42 MB  100.00% 0s4sss
Provisioning OpenShift via 'C:\Users\cdk\.minishift\cache\oc\v1.3.1\oc.exe [cluster up --use-existing-config --host-config-dir /var/lib/minishift/openshift.local.config --host-data-dir /var/lib/minishift/hostdata]'
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... OK
-- Checking for openshift/origin:v1.3.1 image ...
   Pulling image openshift/origin:v1.3.1
   Pulled 0/3 layers, 3% complete
   Pulled 1/3 layers, 34% complete
   Pulled 2/3 layers, 86% complete
   Pulled 3/3 layers, 100% complete
   Extracting
   Image pull complete
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ... OK
-- Checking type of volume mount ...
   Using nsenter mounter for OpenShift volumes
-- Creating host directories ... OK
-- Finding server IP ...
   Using 192.168.99.100 as the server IP
-- Starting OpenShift container ...
   Creating initial OpenShift configuration
   Starting OpenShift using container 'origin'
   Waiting for API server to start listening
   OpenShift server started
-- Installing registry ... OK
-- Installing router ... OK
-- Importing image streams ... OK
-- Importing templates ... OK
-- Login to server ... OK
-- Creating initial project "myproject" ... OK
-- Server Information ...
   OpenShift server started.
   The server is accessible via web console at:
       https://192.168.99.100:8443

   You are logged in as:
       User:     developer
       Password: developer

   To login as administrator:
       oc login -u system:admin

$ minishift.exe stop
Stopping local OpenShift cluster...
Error unregistring machine: Something went wrong running an SSH command!
command : sudo cat /etc/os-release
err     : exit status 255
output  :

cdk@WIN10 ~/minishift1.0beta2
$ minishift.exe ip
E0124 06:27:45.217257    5096 ip.go:45] Error getting IP:  Something went wrong running an SSH command!
command : ip addr show
err     : exit status 255
output  :

cdk@WIN10 ~/minishift1.0beta2
$ minishift.exe ssh
E0124 06:28:24.376632    3016 ssh.go:39] Error attempting to ssh into machine:  exit status 255

cdk@WIN10 ~/minishift1.0beta2
$ minishift.exe status
Running
```

**Comment by hferentschik** [comment]
@amitkrout, how is this exactly related?

**Comment by amitkrout** [comment]
@hferentschik Me and @praveenkumar had the discussion on the same issue which i was facing as my above comment. Even i am not sure how it was exactly related to this but its something related to "unregistering machine". If you think the observation content i added to this issue is not related to then we can create a new issue on the same.

**Comment by hferentschik** [comment]
Resolved by pull request #384.
Will be followed up by issue #396.
